pages/api/v1/fantasy/index.ts

```diff
@@ -0,0 +1,73 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import SQL from 'sql-template-strings';
+import Cors from 'cors';
+import { query } from '../../../../lib/db';
+import use from '../../../../lib/middleware';
+
+const cors = Cors({
+  methods: ['GET', 'HEAD'],
+});
+
+export interface FantasyData {
+  seasonID: number;
+  position: string;
+  name: string;
+  fantasyPoints: number;
+}
+
+export default async (
+  req: NextApiRequest,
+  res: NextApiResponse<Array<FantasyData> | string>
+): Promise<void> => {
+  await use(req, res, cors);
+
+  const { season, position, name } = req.query;
+
+  const [seasonid] =
+    (!Number.isNaN(+season) && [{ SeasonID: +season }]) ||
+    (await query(SQL`
+      SELECT DISTINCT SeasonID
+      FROM player_master
+      WHERE LeagueID=0
+      ORDER BY SeasonID DESC
+      LIMIT 1
+  `));
+
+  const search = SQL`SELECT * FROM fantasypoints WHERE seasonID=${seasonid.SeasonID}`
+    .append(
+      position
+        ? SQL`
+         AND position=${position}
+        `
+        : ''
+    ).append(
+      name
+        ? SQL`
+         AND name=${name}
+        `
+        : ''
+    );
```

Comment:
These ternaries may have problems if somebody enters something that could appear as "falsy" like `0` or `false`. Can we add back the checks for if it is not `null` since that should cover what we are looking for (and picks up `undefined`) sorry for the confusion

Reply:
one of these times I will understand what you want from me the first time lol